kernel/src/main/java/org/cristalise/kernel/lifecycle/instance/predefined/RemoveC2KObject.java

```diff
@@ -32,7 +33,9 @@
 public class RemoveC2KObject extends PredefinedStep {
     public RemoveC2KObject() {
         super();
-        getProperties().put("Agent Role", "Admin");
+        if (Gateway.getProperties().getBoolean("PredefinedStep.AgentRole.enableAdmin", false)) {
+            getProperties().put(AGENT_ROLE.getName(), ADMIN_ROLE.getName());
+        }

```

Comment:
In order to reduce code duplication, move this code into `PredefinedStep.java`. It should be in a new protected method (maybe called `setAgentRole()`?) that is invoked from the constructor of the `PredefinedStep` class. It can be overridden in `Erase` to include the extra roles.

Reply:
Done